In [3]:
import pandas as pd
df=pd.read_csv('/Users/cyrilmahfouz/Desktop/SEF/Week2/Pandas_Rfinal/Telco-Customer-Churn.csv')
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7043 entries, 0 to 7042
Data columns (total 21 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   customerID        7043 non-null   object 
 1   gender            7043 non-null   object 
 2   SeniorCitizen     7043 non-null   int64  
 3   Partner           7043 non-null   object 
 4   Dependents        7043 non-null   object 
 5   tenure            7043 non-null   int64  
 6   PhoneService      7043 non-null   object 
 7   MultipleLines     7043 non-null   object 
 8   InternetService   7043 non-null   object 
 9   OnlineSecurity    7043 non-null   object 
 10  OnlineBackup      7043 non-null   object 
 11  DeviceProtection  7043 non-null   object 
 12  TechSupport       7043 non-null   object 
 13  StreamingTV       7043 non-null   object 
 14  StreamingMovies   7043 non-null   object 
 15  Contract          7043 non-null   object 
 16  PaperlessBilling  7043 non-null   object 


<h3><font color='red'>Introduction</font></h3>

In [26]:
df.head()


customerID          7043
gender              7043
SeniorCitizen       7043
Partner             7043
Dependents          7043
tenure              7043
PhoneService        7043
MultipleLines       7043
InternetService     7043
OnlineSecurity      7043
OnlineBackup        7043
DeviceProtection    7043
TechSupport         7043
StreamingTV         7043
StreamingMovies     7043
Contract            7043
PaperlessBilling    7043
PaymentMethod       7043
MonthlyCharges      7043
TotalCharges        7043
Churn               7043
dtype: int64

To start we are looking at a dataframe representing the customers in a telecominication company .This dataframe contains information about 7043 customer.Furthermore we can detect that all of these columns revolves around one target value and that is whether the customer churn or not.<br>
Let's begin by identifying the interesting columns and how could they possibly relate to the churning of the customer. The logical links that affect the prices are:<br>
1. The tenure.<br>
2. The age range (Senior or not)
3. Whether or not the customer is dependant on somebody or not.(Partners).<br>
4. What is the type of service that he is using.<br>
5. The type of contract that he is enrolled with.<br>
6. The total charges that he has to pay. 

<h3><font color='red'>Anazlyze</font></h3>

To begin with let's understand the meaning of churn and why it is that important. The churn column represents whether the custumer bail on the company or not. So we could also get from this column if the customer is loyal towards the company.<br>
After understandig this, lets kick it by getting the total number of churn and analyzing the tenure of each customer.

In [38]:
total_churn=df.loc[df.Churn=='Yes']['customerID'].count()
print("the total number of churn is: ",total_churn)

the total number of churn is:  1869


<h4><b>The tenure</b></h4>

The tenure of a customer refers to the amount of time a customer has been using or subscribed to a particular service.Lets take a deeper look into it:

In [29]:
avrg_tenure=df['tenure'].mean().round()
print("the average tenure duration is: ",avrg_tenure)
print("the maximum tenure duration is: ",df['tenure'].max())
print()

#lets group them into 2 groups: 
df_=df.copy()
df_['above_avrg_tenure']=df_['tenure'].apply(lambda x:1 if x>avrg_tenure else 0)


print("the number of customer above avrg tenure: ",df_.loc[df_['above_avrg_tenure']== 1]['customerID'].count())
print("the number of customer lower than avrg tenure: ",df_.loc[df_['above_avrg_tenure']== 0]['customerID'].count())
print()

print("number of churn for customer having above avrg tenure: ",df_.loc[(df_['above_avrg_tenure'] == 1) & (df_['Churn'] == 'Yes')]['Churn'].count())
print('number of churn for customer having lower avrg tenure: ',df_.loc[(df_['above_avrg_tenure'] == 0) & (df_['Churn'] == 'Yes')]['Churn'].count())


the average tenure duration is:  32.0
the maximum tenure duration is:  72

the number of customer above avrg tenure:  3268
the number of customer lower than avrg tenure:  3775

number of churn for customer having above avrg tenure:  409
number of churn for customer having lower avrg tenure:  1460


First of all we started the analysis by getting the average tenure duration of the customers and we got the number 32.<br>Then we proceeded by grouping the customers into 2 groups. The first group represents the customer with an above average tenure duration and the second group represents customer with a lower average tenure duration.<br>
<br>
It appears that the number of customer that are in the first group are 3268 person and the number of customer that are in the second group are 3775 person;thise 2 numbers are very close to each other which is great for determining whether or not the group has a tendency to churn.<br>
<br>
We carry on by checking both groups number of churn customer and here is where it gets interesting.Even though the number of customer in each group is close, we remark that the number of customer that churn and have  an above avrg tenure is 409 and the number of churn for customer having lower average tenure is 1460<br>
<br><font color='"#FFCCCB"'>In conclusion, it's evident that customers with above-average tenure show a lower churn rate compared to customers with below-average tenure. This observation highlights the significant impact of tenure on customer churn within the provided dataset.</font><br>
<br>

<h4><b>Age range</b></h4>

In [54]:
senior=df_.loc[df_.SeniorCitizen==1]['customerID'].count()
junior=7043-senior
print("the number of senior citizen customer is: ",senior)
print("the number of junior citizen customer is: ",junior)
print()

senior_churn=df_.loc[(df_.SeniorCitizen==1) & (df_.Churn=='Yes')]['customerID'].count()
junior_churn=df_.loc[(df_.SeniorCitizen==0) & (df_.Churn=='Yes')]['customerID'].count()
print("the total number of senior that bailed: ",senior_churn)
print("the total number of junior thay bailed: ",junior_churn)
print()


perc_senior_custnb=(senior/7043)*100
perc_junior_custnb=100-perc_senior_custnb
print("the percentage of senior citizen that are customers is: ",perc_senior_custnb.round(2))
print("the percentage of junior citizen that are customers is: ",perc_junior_custnb.round(2))
print()

perc_senior_bail=(senior_churn/total_churn)*100
perc_junior_bail=100-perc_senior_bail
print("the percentage of the customer that bailed and is a senior citizen is: ",perc_senior_bail.round(2))
print("the percentage of the customer that bailed and is a junior citizen is: ",perc_junior_bail.round(2))

the number of senior citizen customer is:  1142
the number of junior citizen customer is:  5901

the total number of senior that bailed:  476
the total number of junior thay bailed:  1393

the percentage of senior citizen that are customers is:  16.21
the percentage of junior citizen that are customers is:  83.79

the percentage of the customer that bailed and is a senior citizen is:  25.47
the percentage of the customer that bailed and is a junior citizen is:  74.53


Concerning the age range of the customer, we began by grouping the data into 2 groups;senior citizen and junior citizen<br>
<br>The number of senior citizen present in our data is 1142 and the number of junior citizen is 5901.These numbers seems far from each other but there is something intreaguing about them.<br>
<br>Even though the percentage representing the senior customer is only 16.21% ,the percentage representing the customer that are senior and became unloyal is 25% which is a quarter of the churn column.<br>
<br><font color='"#FFCCCB"'>So, even though the number of senior citizen is low, the percentage of bail is not.It could be affected by a lot of circumstances and maybe they rely on financial support of somebody else. Lets tackle the dependencie section<br>
<br>
